File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.java
//apna college article bruteforce
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n = grid.length;
        int N = n * n;
        int[] freq = new int[N + 1]; // Frequency array
        // Initialise repeated and missing with -1
        int repeated = -1, missing = -1;

        // Count occurrences of each number in the grid
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                freq[grid[i][j]]++;
            }
        }

        // Find the repeated and missing numbers
        for (int i = 1; i <= N; i++) {
            if (freq[i] == 2) {
                repeated = i;
            } else if (freq[i] == 0) {
                missing = i;
            }
        }

        return new int[] { repeated, missing };
    }
}

File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.py
# apna college article
from typing import List


class Solution(object):
   def findMissingAndRepeatedValues(self, grid):
       """
       :type grid: List[List[int]]
       :rtype: List[int]
       """
       n = len(grid)
       n2 = n * n
       freq = [0] * (n2 + 1)  # Frequency array to track occurrences
     # Initialise repeated and missing with -1


       repeated = -1
       missing = -1


       # Count occurrences of each number in the grid
       for i in range(n):
           for j in range(n):
               freq[grid[i][j]] += 1


       # Find the repeated and missing numbers
       for i in range(1, n2 + 1):
           if freq[i] == 2:
               repeated = i
           elif freq[i] == 0:
               missing = i


       return [repeated, missing]






